compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -2393,6 +2359,164 @@ public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
+        BIRVariableDcl typedescVar = getTypedescVariable(type);
+        if (typedescVar != null) {
+            return new BIRNonTerminator.NewStructure(pos, toVarRef, new BIROperand(typedescVar), fields);
+        } else {
+            createNewTypedescInst(pos, type);
+            return new BIRNonTerminator.NewStructure(pos, toVarRef, this.env.targetOperand, fields);
+        }
+    }
+
+    public String getTypedescFieldName(String name) {
+        return "$typedesc$" + name;
+    }
+
+    private BIRVariableDcl getTypedescVariable(BType type) {
+        BIRVariableDcl variableDcl = findInPackageScope(type);
+        if (variableDcl != null) {
+            return variableDcl;
+        }
+
+        variableDcl = findInLocalSymbolVarMap(type, env.symbolVarMap);
+        if (variableDcl != null) {
+            return variableDcl;
+        }
+
+        return findInGlobalSymbolVarMap(type, this.globalVarMap);
+    }
+
+    private BIRVariableDcl findInPackageScope(BType type) {
+        BTypeSymbol typeSymbol = type.tsymbol;
+        if (typeSymbol != null && isDifferentPackage(type)) {
+            BPackageSymbol packageSymbol = (BPackageSymbol) typeSymbol.owner;
+            Scope.ScopeEntry scopeEntry =
+                                    packageSymbol.scope.lookup(new Name(getTypedescFieldName(typeSymbol.name.value)));
+            BSymbol symbol = scopeEntry.symbol;
+            if (symbol != null) {
+                BLangPackageVarRef packageVarRef = createPackageVarRef(symbol);
+                visit(packageVarRef);
+                return this.env.targetOperand.variableDcl;
+            }
+        }
+        return null;
+    }
+
+    private boolean isDifferentPackage(BType type) {
+        return type.tsymbol.owner.tag == SymTag.PACKAGE && !type.tsymbol.pkgID.equals(env.enclPkg.packageID);
+    }
+
+    private BLangPackageVarRef createPackageVarRef(BSymbol symbol) {
+        BLangPackageVarRef packageVarRef = new BLangPackageVarRef((BVarSymbol) symbol);
+        packageVarRef.pos = symbol.pos;
+        packageVarRef.setBType(symbol.getType());
+        return packageVarRef;
+    }
+
+    private BIRVariableDcl findInLocalSymbolVarMap(BType type, Map<BSymbol, BIRVariableDcl> varMap) {
+        for (Map.Entry<BSymbol, BIRVariableDcl> entry : varMap.entrySet()) {
+            if (isTypeDescSymbol(entry.getKey(), Types.getImpliedType(type))) {
+                return varMap.get(entry.getKey());
+            }
+        }
+        return null;
+    }
+
+    private BIRVariableDcl findInGlobalSymbolVarMap(BType type, Map<BSymbol, BIRGlobalVariableDcl> varMap) {
+        for (Map.Entry<BSymbol, BIRGlobalVariableDcl> entry : varMap.entrySet()) {
+            if (isTypeDescSymbol(entry.getKey(), Types.getImpliedType(type))) {
+                return varMap.get(entry.getKey());
+            }
+        }
+        return null;
+    }
```

Comment:
This method is a duplication of `findInLocalSymbolVarMap`. We can pass the `globalVarMap` to the `findInLocalSymbolVarMap`. Then we can rename the function to the `findInSymbolVarMap`.